### PR TITLE
Refactor GetUserByUsername method not to suppress an error

### PR DESCRIPTION
**Pull request by shuheiktgw**
## Overview
`GetUserByUsername` method in `user.go` suppresses an error and that makes it really hard to use the method. I believe the method itself should focus on fetching a user by name, and a client who uses the method should decided how to handle the error the method returns.

So I refactored the method not to suppress an error, and simply return it.

Thank you! 👍 

# FYI
Once this PR is merged, some tests in another PR I made (https://github.com/gophish/gophish/pull/919) should be broken, so I'll fix them after this PR is merged.

### CODE REVIEW

File: models/models_test.go [hunk 1]
```diff
@@ -100,6 +100,18 @@ func (s *ModelsSuite) TestGetUser(c *check.C) {
 	c.Assert(u.Username, check.Equals, "admin")
 }
 
+func (s *ModelsSuite) TestGetUserByUsernameWithExistedUser(c *check.C) {
```
Comment (jordan-wright) [inline models/models_test.go]: Super minor nitpick, can we change these to "WithExisting" instead of "WithExisted"?

**Review by jordan-wright — CHANGES_REQUESTED** [review]
I agree with this. You're right - it's almost certainly better practice to just pass the error up and let the upper layer decide how to handle it.

One minor nitpick with the test names and I'll be happy to merge this in.

Thanks for sending this over!

File: models/models_test.go (hunk 1 above)
Comment (shuheiktgw) [reply models/models_test.go]: Thank you for your review! 
I just fixed the names, so please let me know if you have any comments on this PR! 👍

**Review by jordan-wright — APPROVED** [review]

**Comment by jordan-wright** [comment]
No idea what happened with the odd indentation in `models_test.go` when resolving the merge. Oh well, it'll get fixed in the next fmt when I change that file again. The code still looks good, so I'll merge it in.

Thanks again for sending this in!

**Pull request merged**